beamer/chains.py

```diff
@@ -0,0 +1,82 @@
+from dataclasses import dataclass
+from typing import Any
+
+from beamer.typing import ChainId
+
+
+@dataclass(frozen=True)
+class ChainDescriptor:
+    id: ChainId
+    ecosystem_name: str
+    name: str
+    type2: bool = False
+    bedrock: bool = False
+    testnet: bool = False
+    local: bool = False
+
+
+DESCRIPTORS = (
+    # Ethereum
+    ChainDescriptor(id=ChainId(1), ecosystem_name="ethereum", name="mainnet", type2=True),
+    ChainDescriptor(
+        id=ChainId(5), ecosystem_name="ethereum", name="goerli", type2=True, testnet=True
+    ),
+    ChainDescriptor(
+        id=ChainId(1337), ecosystem_name="ethereum", name="local", type2=True, local=True
+    ),
+    # Optimism
+    ChainDescriptor(
+        id=ChainId(10), ecosystem_name="optimism", name="mainnet", type2=True, bedrock=True
+    ),
+    ChainDescriptor(
+        id=ChainId(420),
+        ecosystem_name="optimism",
+        name="goerli",
+        type2=True,
+        bedrock=True,
+        testnet=True,
+    ),
+    ChainDescriptor(
+        id=ChainId(901),
+        ecosystem_name="optimism",
+        name="local",
+        type2=True,
+        bedrock=True,
+        local=True,
+    ),
+    # Arbitrum
+    ChainDescriptor(id=ChainId(42161), ecosystem_name="arbitrum", name="mainnet"),
+    ChainDescriptor(id=ChainId(421613), ecosystem_name="arbitrum", name="goerli", testnet=True),
+    ChainDescriptor(id=ChainId(412346), ecosystem_name="arbitrum", name="local", local=True),
+    # Polygon ZkEVM
+    ChainDescriptor(id=ChainId(1101), ecosystem_name="polygon_zkevm", name="mainnet"),
+    ChainDescriptor(id=ChainId(1442), ecosystem_name="polygon_zkevm", name="goerli", testnet=True),
+    ChainDescriptor(id=ChainId(1001), ecosystem_name="polygon_zkevm", name="local", local=True),
+    # Base
+    ChainDescriptor(
+        id=ChainId(84531), ecosystem_name="base", name="goerli", testnet=True, bedrock=True
+    ),
+)
+
+
+def _prepare_descriptors_for_lookup() -> dict[ChainId, ChainDescriptor]:
+    descriptors = {}
+    for descriptor in DESCRIPTORS:
+        descriptors[descriptor.id] = descriptor
+    return descriptors
+
+
+_CHAIN_DESCRIPTORS = _prepare_descriptors_for_lookup()
```

Comment:
We don't actually need to have both the tuple and the dict keyed by the chain ID, it is sufficient to only have the latter. The `search` function can then just iterate over dict's `values()` and we get  the same behaviour as if we're iterating over the tuple.

Reply:
but you said tuple first :D

Reply:
Right and it's totally fine to have tuple as an input as it is now, my comment above is about the fact that we have two module-level objects with the same data, just in different form. So what I was trying to say is, we only need one of those two -- the dict which is a result of the prepare function.

I meant something like
```
_DESCRIPTORS = (...)   # tuple
_DESCRIPTORS = _prepare(_DESCRIPTORS)  # now it's a dict and we only have one copy of the data
```
which could even be done as
```
_DESCRIPTORS = _prepare(
   ChainDescriptor(...),
   ChainDescriptor(...)
...
)
```

Reply:
made it dictionary directly without any tuple now